imagepuller/internal/service/verify.go

```diff
@@ -0,0 +1,84 @@
+// Copyright 2025 Edgeless Systems GmbH
+// SPDX-License-Identifier: BUSL-1.1
+
+package service
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log/slog"
+	"strings"
+
+	"github.com/google/go-containerregistry/pkg/name"
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+	"github.com/google/go-containerregistry/pkg/v1/remote/transport"
+)
+
+func (s *ImagePullerService) getAndVerifyImage(ctx context.Context, log *slog.Logger, imageURL string) (v1.Image, error) {
+	ref, err := name.ParseReference(imageURL)
+	if err != nil {
+		return nil, fmt.Errorf("parsing the image URL as a reference: %w", err)
+	}
+
+	digestExpectedRaw, err := name.NewDigest(imageURL)
+	if err != nil {
+		return nil, fmt.Errorf("parsing image digest: %w", err)
+	}
+	digestExpected := digestExpectedRaw.DigestStr()
```

Comment:
`name.Digest` implements `name.Reference` - I think we could skip the `name.ParseReference` here.

```suggestion
	ref, err := name.NewDigest(imageURL)
	if err != nil {
		return nil, fmt.Errorf("parsing image digest: %w", err)
	}
	digestExpected := ref.DigestStr()
```